Makefile

```diff
@@ -92,3 +92,31 @@ oam-runtime.help:
 help-special: oam-runtime.help
 
 .PHONY: oam-runtime.help help-special
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# find or download controller-gen
+# download controller-gen if necessary
+controller-gen:
```

Comment:
I like your pattern of just adding `/crds` to the `.gitignore`, I can re-enable the generation here and we can just ignore the output: https://github.com/crossplane/oam-kubernetes-runtime/pull/6/files#diff-26d5f9a9604b23990843fe2c336219caR23

We already pin `controller-gen` in `generate.go` so no need to add this portion to the `Makefile`.